src/DotNet.Status.Web/Controllers/TelemetryController.cs

```diff
@@ -0,0 +1,78 @@
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Microsoft.DotNet.Kusto;
+using Kusto.Ingest;
+using System.Collections.Generic;
+
+namespace DotNet.Status.Web.Controllers
+{
+    [Route("api/telemetry")]
+    [ApiController]
+    public class TelemetryController : ControllerBase
+    {
+        private readonly ILogger<TelemetryController> _logger;
+        private readonly IOptionsSnapshot<TelemetryOptions> _options;
+        private readonly IKustoIngestClient _client;
+
+        public TelemetryController(
+            ILogger<TelemetryController> logger,
+            IOptionsSnapshot<TelemetryOptions> options,
+            IKustoIngestClient client = null)
+        {
+            _logger = logger;
+            _options = options;
+            _client = client;
+        }
+
+        [HttpPost("collect/ArcadeValidation")]
+        public async Task<IActionResult> CollectArcadeValidation([Required] ArcadeValidationData data)
+        {
+            _logger.LogInformation("Start Collect Arcade Validation data");
+
+            TelemetryOptions options = _options.Value;
+
+            if(null == options)
+            {
+                _logger.LogError("TelemetryOptions were not loaded.");
+                return StatusCode(500);
+            }
+
+            if (null == _client && string.IsNullOrEmpty(options.KustoIngestConnectionString))
+            {
+                _logger.LogError("No KustoIngestConnectionString set");
```

Comment:
You could just "throw" some exception here. Asp.NET will both track the exception and 500 the request for you at that point. :-) (And it's also easier to write a unit test for)

Reply:
I can just throw the exception and not worry about logging it, right?

Reply:
Yup. AI/ASP.NET should already have hooked up something that should result in a "TrackException" call with the right stuff in it.

If it was an expected code path, I'd say don't do that (since exceptions in expected code paths make debugging hard), but in normal operation, this exception shouldn't throw , so it's fine to just fire it and let the system report it.